### PR TITLE
fix(iis): deep merge configs

### LINKED ISSUE #2350: deep merge IIS config
### Environment
------------------------------
- Operating System: Windows_NT
- Node Version:     v20.12.0
- Nuxt Version:     3.11.2
- CLI Version:      3.11.1
- Nitro Version:    2.9.6
- Package Manager:  pnpm@8.4.0
------------------------------

### Reproduction

[/src/presets/iis.ts#L152](https://github.com/unjs/nitro/blob/main/src/presets/iis.ts#L152)

### Describe the bug

e.g.
```{...{ configuration: { 'system.webServer': [ 'a' ] } }, ...{ configuration: { appSettings: [ 'b' ] } }}```
 output:
```{ configuration: { appSettings: [ 'b' ] }```

use Lodash

```
_.merge({ configuration: { 'system.webServer': [ 'a' ] } }, { configuration: { appSettings: [ 'b' ] } })
```
output:
```
{"configuration":{"system.webServer":["a"],"appSettings":["b"]}}
```

### Additional context

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by Ariesly**
### 🔗 Linked issue

Closes #2350

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

deep merge use ```defu```

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/presets/iis.ts [hunk 1]
```diff
@@ -116,7 +117,7 @@ async function iisnodeXmlTemplate(nitro: Nitro) {
       await parseXmlDoc(fileString);
 
     if (nitro.options.iis.mergeConfig && !nitro.options.iis.overrideConfig) {
-      return buildNewXmlDoc({ ...originalWebConfig, ...fileWebConfig });
+      return buildNewXmlDoc(defu(originalWebConfig, fileWebConfig));
```
Comment (pi0) [inline src/presets/iis.ts:120]:
```suggestion
      return buildNewXmlDoc(defu(fileWebConfig, originalWebConfig));
```

File: src/presets/iis.ts [hunk 2]
```diff
@@ -149,7 +150,7 @@ async function iisXmlTemplate(nitro: Nitro) {
...
```
Comment (pi0) [inline src/presets/iis.ts:153]:
```suggestion
      return buildNewXmlDoc(defu(fileWebConfig, originalWebConfig));
```

**Review by pi0 — APPROVED** [review]
Thanks!

**Comment by Ariesly** [comment]
@pi0 Can you help merge these? The current CI timeout error was not caused by this merge.

**Pull request merged**